src/core/lib/security/transport/server_auth_filter.c

```diff
@@ -242,7 +242,6 @@ static grpc_error *init_channel_elem(grpc_exec_ctx *exec_ctx,
 
   GPR_ASSERT(!args->is_last);
   GPR_ASSERT(auth_context != NULL);
-  GPR_ASSERT(creds != NULL);
 
```

Comment:
Just sanity checking, but it's ok for a NULL creds to be used below:
chand->creds = grpc_server_credentials_ref(creds);
?